catalog/service/common/src/main/java/org/projectnessie/catalog/service/objtypes/SignerKey.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.catalog.service.objtypes;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import java.time.Instant;
+import javax.crypto.spec.SecretKeySpec;
+import org.immutables.value.Value;
+import org.projectnessie.nessie.immutables.NessieImmutable;
+
+@NessieImmutable
+@JsonSerialize(as = ImmutableSignerKey.class)
+@JsonDeserialize(as = ImmutableSignerKey.class)
+public interface SignerKey {
+  String name();
+
+  String secretKey();
+
+  Instant creationTime();
+
+  Instant rotationTime();
+
+  Instant expirationTime();
+
+  @JsonIgnore
+  @Value.Lazy
+  default SecretKeySpec secretKeySpec() {
+    byte[] secretKeyBytes = ("NessieSign:" + secretKey()).getBytes(UTF_8);
```

Comment:
This looks suspicious, shouldn't we decode `secretKey()` (it's base64-encoded) first? And why prepend `"NessieSign:"`?